src/addons/Portal/Portal.js

```diff
@@ -229,6 +273,12 @@ class Portal extends Component {
     this.trySetState({ open: false })
   }
 
+  closeWithTimeout = (e, delay) => {
+    // React wipes certain props (e.g. currentTarget) so we need to clone.
+    const eventClone = { ...e }
```

Comment:
Tried `e.persist` initially but doesn't preserve the currentTarget, see https://github.com/facebook/react/issues/2857. This is kinda gross but it's the only way to make a copy of the event as-is.